ui/src/components/segments/ConstraintForm.tsx

```diff
@@ -214,6 +218,22 @@ function ConstraintValueDateTimeInput(props: ConstraintInputProps) {
   );
 }
 
+const validationSchema = Yup.object({
+  property: requiredValidation
+})
+  .when({
+    is: (c: IConstraint) =>
+      c.type === ConstraintType.STRING &&
+      ['isoneof', 'isnotoneof'].includes(c.operator),
+    then: (schema) => schema.shape({ value: jsonStringArrayValidation })
+  })
+  .when({
+    is: (c: IConstraint) =>
+      c.type === ConstraintType.NUMBER &&
+      ['isoneof', 'isnotoneof'].includes(c.operator),
+    then: (schema) => schema.shape({ value: jsonNumberArrayValidation })
+  });
+
```

Comment:
looks great, in a future PR it might be nice if we could put in some placeholder text for value when they select 'IS ONE OF' or 'IS NOT ONE OF', maybe have it change between strings and numbers depending on if the type is String or Number?

![CleanShot 2023-11-12 at 16 50 23](https://github.com/flipt-io/flipt/assets/209477/8cccd850-aff0-434c-9dd2-2e90999aa056)

something like ☝🏻 but `value` input text would be greyed out as placeholder text to show the shape expected